library/network/src/modules/SuSEFirewallServices.rb

```diff
@@ -745,6 +766,181 @@ def GetPossiblyConflictServices
     publish function: :GetPossiblyConflictServices, type: "list <string> ()"
   end
 
-  SuSEFirewallServices = SuSEFirewallServicesClass.new
-  SuSEFirewallServices.main
+  class SuSEFirewalldServicesClass < SuSEFirewallServicesClass
+    include Yast::Logger
+
+    SERVICES_DIR = ["/etc/firewalld/services", "/usr/lib/firewalld/services"]
+
+    IGNORED_SERVICES = ["..", "."]
+
+    def initialize
+      @services = nil
+
+      @known_services_features = {
+        "TCP"     => "tcp_ports",
+        "UDP"     => "udp_ports",
+        "IP"      => "ip_protocols",
+        "MODULES" => "modules"
+      }
+
+      @known_metadata = { "Name" => "name", "Description" => "description" }
+
+      # firewall needs restarting. Always false for firewalld
+      @sfws_modified = false
+    end
+
+    # Reads definition of services that can be used in FirewallD
+    #
+    # @return [Boolean] if successful
+    def ReadServicesDefinedByRPMPackages
+      log.info "Reading FirewallD services from"
+
+      service_attrs = ["modules", "ports", "protocols"]
+
+      @services ||= {}
+
+      # We need to start the backend first
+      SuSEFirewall.StartServices() unless SuSEFirewall.IsStarted()
```

Comment:
That is a non-obvious side effect. If it cannot be avoided, please
- document it
- make sure it leaves a trace in the log